src/utils.ts

```diff
@@ -476,24 +484,32 @@ export class Utils {
     }
   }
 
-  public static initEvent<T>(e: DragEvent | MouseEvent, info: { type: string; target?: EventTarget }): T {
-    const evt = { type: info.type };
-    const obj = {
-      button: 0,
-      which: 0,
-      buttons: 1,
-      bubbles: true,
-      cancelable: true,
-      target: info.target ? info.target : e.target
-    };
-    // don't check for `instanceof DragEvent` as Safari use MouseEvent #1540
-    if ((e as DragEvent).dataTransfer) {
-      evt['dataTransfer'] = (e as DragEvent).dataTransfer; // workaround 'readonly' field.
+    public static initEvent<T extends DragEvent | MouseEvent>(e: DragEvent | MouseEvent, info: { type: string; target?: EventTarget }): GridStackDragMouseEvent<T> {
+        return {
+            type: info.type,
+            button: 0,
+            buttons: 1,
+            bubbles: true,
+            cancelable: true,
+            target: info?.target || e.target,
+            ...((e as DragEvent).dataTransfer && {
+                dataTransfer: (e as DragEvent).dataTransfer
+            } as any),
+            // keys
+            altKey: e.altKey,
+            ctrlKey: e.ctrlKey,
+            metaKey: e.metaKey,
+            shiftKey: e.shiftKey,
+            // point info
+            pageX: e.pageX,
+            pageY: e.pageY,
+            clientX: e.clientX,
+            clientY: e.clientY,
+            screenX: e.screenX,
+            screenY: e.screenY,
+            originalEvent: e,

```

Comment:
so this IS the only real change you made. not clear to me what you have is cleaner - definitely more verbose.

Reply:
yeah that's why I did 2 commits to try to separate the real change to the reformat.
I closed the PR for now because I don't have the correct typings.

Reply:
what's your thought on how to implement this `originalEvent: e` ?

Reply:
as I asked in he bug posted, I need to understand why it is needed... never been an issue so trying to udnerstand the use case.